src/application/services/axisValInputHandler/axisValInputHandler.ts

```diff
@@ -0,0 +1,227 @@
+      originalVal.startsWith('1e')
+    ) {
+      return MathUtils.decrementExponent(originalVal)
+    }
+
+    throw new Error('Cannot multiply this value by 10')
+  }
+
+  handleMultiplyAxisValue(axisSetId: number, axisKey: AxisKey): void {
+    const multipliedVal = this.getInputValueMultipliedByTen(axisSetId, axisKey)
+
+    this.setInputValue(axisSetId, axisKey, multipliedVal)
+    this.setConvertedAxisValToAxisSet(axisSetId, axisKey)
+  }
+
+  handleDivideAxisValue(axisSetId: number, axisKey: AxisKey): void {
+    const multipliedVal = this.getInputValueDividedByTen(axisSetId, axisKey)
+
+    this.setInputValue(axisSetId, axisKey, multipliedVal)
+    this.setConvertedAxisValToAxisSet(axisSetId, axisKey)
+  }
+
+  private validateInputValues(axisSetId: number): void {
+    const axisSet = this.axisSetRepository.activeAxisSet
+    const { x1, x2, y1, y2 } = this.getAxisSetInputValues(axisSetId)
+
+    const isInvalidInputFormat = {
+      x1: this.getInputValueFormat(x1) === 'invalidFormat',
+      x2: this.getInputValueFormat(x2) === 'invalidFormat',
+      y1: this.getInputValueFormat(y1) === 'invalidFormat',
+      y2: this.getInputValueFormat(y2) === 'invalidFormat',
+    }
+
+    const xValuesAreSame = x1 !== '0' && x2 !== '0' && x1 === x2
+    const isXLogScaleAndZero = axisSet.xIsLogScale && (x1 === '0' || x2 === '0')
+    const yValuesAreSame = y1 !== '0' && y2 !== '0' && y1 === y2
+    const isYLogScaleAndZero = axisSet.yIsLogScale && (y1 === '0' || y2 === '0')
+
+    this.validationStatus[axisSetId] = {
+      isInvalidInputFormat,
+      xValuesAreSame,
+      isXLogScaleAndZero,
+      yValuesAreSame,
+      isYLogScaleAndZero,
+    }
+  }
+
+  private getValidationStatus(
+    axisSetId: number,
+  ): AxisValuesInputValidationStatus {
+    this.validateInputValues(axisSetId)
+    return (
+      this.validationStatus[axisSetId] || {
+        isInvalidInputFormat: {
+          x1: false,
+          x2: false,
+          y1: false,
+          y2: false,
+        },
+        xValuesAreSame: false,
+        isXLogScaleAndZero: false,
+        yValuesAreSame: false,
+        isYLogScaleAndZero: false,
+      }
+    )
+  }
+
+  getValidationMessage(axisSetId: number): string {
+    let resultMessage: string = ''
+    const validationStatus = this.getValidationStatus(axisSetId)
+
+    for (const [axisKey, isInvalid] of Object.entries(
+      validationStatus.isInvalidInputFormat,
+    )) {
+      if (isInvalid) {
+        resultMessage += `${axisKey} value is invalid\n`
+      }
+    }
+
+    if (validationStatus.xValuesAreSame) {
+      resultMessage += 'X1 and X2 should not be the same value\n'
+    }
+
+    if (validationStatus.xValuesAreSame) {
+      resultMessage += 'Y1 and Y2 should not be the same value\n'
+    }
+
+    if (
+      validationStatus.isXLogScaleAndZero ||
+      validationStatus.isYLogScaleAndZero
+    ) {
+      resultMessage += 'In log scale, axis value should not be 0\n'
```

Comment:
I thought it would be better to accumulate the strings into an array and then join them at the end, so you don't have to worry about the trailing newline. 

e.g.

```
  if (isXLogScaleAndZero || isYLogScaleAndZero) {
    messages.push('In log scale, axis value should not be 0')
  }

  return messages.join('\n')
```